### PR TITLE
Sub/feat/fix dev recurring issue

### PR DESCRIPTION
**Pull request by anandpatel504**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -77,7 +77,57 @@ exports.deployment = async (start) => {
...
+    chatService,
+    classesService,
+    userService,
+    displayService,
+    calendarService,
+  } = server.services();
+
+  /* Scheduler- call calendar patch API in every 10 min*/
+  // cron.schedule('00 */10 * * * *', async () => {
+  const [err, nextTwoHoursClasses] = await classesService.getClassesForTwoHours();
```
Comment (saiky76) [inline server/index.js:90]: Why do we need a cron job for this? Can't we schedule it in the application only with a async function?

File: lib/services/classes.js [hunk 2]
```diff
@@ -517,25 +517,104 @@ module.exports = class classesService extends Schmervice.Service {
...
+  /* update google_registration_status*/
+  async updateGRegistrationStatusById(classId) {
     const { ClassRegistrations } = this.server.models();
+    try {
+      await ClassRegistrations.query()
+        .update({ class_id: classId, google_registration_status: true })
+        .where('class_id', classId)
+        .andWhereNot('google_registration_status', null);
+    } catch (err) {
+      return [errorHandler(err), null];
```
Comment (saiky76) [inline lib/services/classes.js:544]: There is absolutely no error logging and debug logging. I do not know why logging is not being taken seriously

File: lib/services/chat.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ module.exports = class ChatService extends Schmervice.Service {
     const client = server.app.chatClient;
     const { sender } = event;
 
-    if (privateRoomId === null) return; // If message is sent to any other room (This also effectively checks if bot is the event sender)
+    if (privateRoomId === null || privateRoomId === 'user_not_found') return; // If message is sent to any other room (This also effectively checks if bot is the event sender)
```
Comment (saiky76) [inline lib/services/chat.js:30]: What is this for?

File: lib/services/classes.js [hunk 4]
```diff
@@ -517,25 +517,104 @@ module.exports = class classesService extends Schmervice.Service {
...
+    }
+  }
 
+  // get classes for a week
+  async getClassesForWeek() {
```
Comment (saiky76) [inline lib/services/classes.js:549]: why do we need classes for one week?

File: server/index.js [hunk 5]
```diff
@@ -77,7 +77,57 @@ exports.deployment = async (start) => {
...
+  const {
+    chatService,
+    classesService,
+    userService,
+    displayService,
+    calendarService,
+  } = server.services();
+
+  /* Scheduler- call calendar patch API in every 10 min*/
+  // cron.schedule('00 */10 * * * *', async () => {
```
Comment (saiky76) [inline server/index.js:89]: Uncomment cron

File: server/index.js [hunk 6]
```diff
@@ -77,7 +77,57 @@ exports.deployment = async (start) => {
...
+        if (regUser.google_registration_status === false) {
+          const [err, classRegUser] = await userService.findById(regUser.user_id);
+          if (!err) {
+            emailList.push({ email: classRegUser.email });
+          }
+          // letting google calendar deciding the previous attendees
+          let regUsers;
+          try {
+            if (c.recurring_id !== null) {
+              regUsers = await calendarService.getCalendarEvent(
```
Comment (saiky76) [inline server/index.js:105]: function should have the right name

File: lib/services/classes.js [hunk 7]
```diff
@@ -517,25 +517,104 @@ module.exports = class classesService extends Schmervice.Service {
...
       return [null, successReg];
     } catch (err) {
       return [errorHandler(err), null];
     }
   }
 
-  async removeRegistrationById(classIds, userId, txn) {
+  /* eslint-disable */
+  /* update google_registration_status*/
+  async updateGRegistrationStatusById(classId) {
```
Comment (saiky76) [inline lib/services/classes.js:536]: No student info here for whom class reg is being set to true

File: lib/services/chat.js (hunk 3 above)
Comment (anandpatel504) [reply lib/services/chat.js:30]: Handled chat room issue.

File: lib/services/classes.js (hunk 4 above)
Comment (anandpatel504) [reply lib/services/classes.js]: Thanks jschankar for the suggestion I have defined a common function.

**Pull request merged**